pkg/ottl/ottlfuncs/func_flatten.go

```diff
@@ -47,52 +49,79 @@ func flatten[K any](target ottl.PMapGetter[K], p ottl.Optional[string], d ottl.O
 		prefix = p.Get()
 	}
 
+	conflict := false
```

Comment:
Can we make this `resolveConflicts` or similar? `conflict` to me makes it sound like there is a conflict, which this doesn't track.